extensions/expression-language/tests/ExpressionLanguageTests.cpp

```diff
@@ -1342,13 +1342,18 @@ TEST_CASE("Reverse DNS lookup with valid timeout parameter", "[ExpressionLanguag
 
   SECTION("Should timeout") {
     auto reverse_lookup_expr_0ms = expression::compile("${reverseDnsLookup(${ip_addr}, 0)}");
-    REQUIRE_NOTHROW(reverse_lookup_expr_0ms(expression::Parameters{flow_file_a}).asString() == "8.8.8.8");
+    std::string reverse_lookup_result = "dns.google";
+    // Occasionally it doesn't time out even with 0ms timeout because it finishes before the timeout-thread starts
+    for (auto number_of_tries = 1; number_of_tries <= 5 && reverse_lookup_result == "dns.google"; ++number_of_tries) {
+      reverse_lookup_result = reverse_lookup_expr_0ms(expression::Parameters{flow_file_a}).asString();
+    }
```

Comment:
If the timeout doesn't happen deterministically, then I'd rather not test it at all in a unit test. I also wouldn't want to short-circuit 0ms to always time out if it wouldn't normally happen, since that would degrade the functionality.

Reply:
That makes sense, at least now this certainly wont fail :sweat_smile: https://github.com/apache/nifi-minifi-cpp/pull/1541/commits/033f20cd28d16fd8c3c535babf06321f6082ab0a